diff --git a/fileutil_linux.go b/fileutil_linux.go
index 370f57e..8793093 100644
--- a/fileutil_linux.go
+++ b/fileutil_linux.go
@@ -33,17 +33,21 @@ func init() {
 
 	tokens := bytes.Split(b, []byte("."))
 	switch len(tokens) {
-	case 3:
-		// Supported since kernel 2.6.38
-		if bytes.Compare([]byte{n(tokens[0]), n(tokens[1]), n(tokens[2])}, []byte{2, 6, 38}) < 0 {
+	case 0:
+		fallthrough
+	case 1:
+		panic("osrelease=" + string(b))
+	case 2:
+		if bytes.Compare([]byte{n(tokens[0]), n(tokens[1])},
+        []byte{2, 7}) < 0 {
 			puncher = func(*os.File, int64, int64) error { return nil }
 		}
-	case 2:
-		if bytes.Compare([]byte{n(tokens[0]), n(tokens[1])}, []byte{2, 7}) < 0 {
+	default: // len(tokens) >= 3
+		// Supported since kernel 2.6.38
+		if bytes.Compare([]byte{n(tokens[0]), n(tokens[1]), n(tokens[2])},
+			[]byte{2, 6, 38}) < 0 {
 			puncher = func(*os.File, int64, int64) error { return nil }
 		}
-	default:
-		panic(n)
 	}
 }
 
